Cargo.toml

```diff
@@ -118,6 +118,10 @@ all-features = true
 # See https://stackoverflow.com/questions/61417452
 rustdoc-args = ["--cfg", "docs_rs"]
 
+[[test]]
+name = "encodings"
+required-features = ["encoding"]
```

Comment:
There are some tests here (which perhaps can be moved to another file) which we still want to run without the encoding feature, e.g. to verify that non-utf8 encodings fail.

However I have been wondering whether it's worth having an encoding feature to begin with going forwards.  Since we have to at least validate that the bytes being read are UTF-8 no matter what, and `encoding_rs` makes that somewhat easier (less brand new code needs to be written and we can delete all this conditional code).

Reply:
Currently all tests in that file under `#[cfg(feature = "encoding")]`. I suppose, that you'll plan to add more later? Or you forget to remove feature gates from some tests?

Reply:
That is currently true, but presuming we do want to have non-decoding reader support that just does validation, we'll want some verified-failure tests for parsing non-UTF-8 encodings.

Reply:
I think, I'll change it now as I've did that, and in the future we always can switch again when we will need that. I'll also move `fuzz` tests to a dedicated file